### DISCUSSION TITLE
Disable predefined snippets per filetype

### DISCUSSION THREAD

**Discussion by manoelpqueiroz** [LazyVim/LazyVim#5153] [Q&A] [answered]
Hi guys, one thing that I have been trying to achieve without success is disabling predefined snippets depending on the filetype (stuff like `c)~` for copyright, `date~`, `uuid~`, `diso~`). As I usually use Neovim as my editor for [`jrnl`](https://github.com/jrnl-org/jrnl), I like to be completely distraction-free when I'm writing an entry.

When LazyVim used `nvim-snippets`, I managed to go around this behaviour with this configuration:

```lua
return {
  "garymjr/nvim-snippets",
  opts = {
    ignored_filetypes = { "*.jrnl", "*.txt" },
    global_snippets = {},
  },
}
```

Obviously, this disabled the snippets for all filetypes at the end of the day, but is was fine with my needs.

Checking the documentation for `blink.cmp`, I have now replaced the previous file with this configuration:

```lua
return {
  "saghen/blink.cmp",
  opts = {
    sources = {
      providers = {
        snippets = {
          opts = {
            ignored_filetypes = {
              "jrnl",
              "txt",
            },
          },
        },
      },
    },
  },
}
```

Yet, the aforementioned snippets still show up when I edit `tst.jrnl` or `tst.txt` after restarting Neovim. I have also tried to use `"*.jrnl"` and `"*.txt"` in the option since the Blink documentation does not specify the patterns to use, but to no avail. I believe this is connected to fine details on how LazyVim declares the snippets.

Could anyone help me shed a light on how to tackle this case? I'd be much appreciated.

**Comment by binhtddev** [comment, marked as answer]
The `ignored_filetypes` config of blink will use the filetype which can be gotten by `set ft?`. With `txt`, I can see it should be `text`, but `jrnl` is empty because it is not recognize by neovim. I suggest using ftdetect like this plugin https://github.com/wren/jrnl.vim/blob/main/ftdetect/jrnl.vim and the final config should be:
```lua
return {
  "saghen/blink.cmp",
  opts = {
    sources = {
      providers = {
        snippets = {
          opts = {
            ignored_filetypes = {
              "jrnl",
              "text",
            },
          },
        },
      },
    },
  },
}
```

**Reply by manoelpqueiroz** [reply]
Wow, so I had an even more basic Neovim configuration under my radar. Thanks a lot for the answer! I'm coming from VS Code so I've been accostumed to using the glob extensions, I never thought Vim itself worked by setting up filetypes which could relate to more than one [file extension](https://neovim.io/doc/user/lua.html#vim.filetype.add()).

I did not even need to use ftdetect, the following already works out of the box if added to `nvim/lua/config/options.lua`:

```lua
vim.filetype.add({ extension = { jrnl = "jrnl" } })
```
